types/koa/index.d.ts

```diff
@@ -513,7 +513,7 @@ declare class Application<
      * Return a request handler callback
      * for node's native http/http2 server.
      */
-    callback(): (req: IncomingMessage | Http2ServerRequest, res: ServerResponse | Http2ServerResponse) => void;
+    callback(): (req: IncomingMessage | Http2ServerRequest, res: ServerResponse | Http2ServerResponse) => Promise<void>;
```

Comment:
`callback` doesn't return a `Promise<void>`, if anything, it returns the `handleRequest` function, which then returns a `Promise<void>`

---

```
Welcome to Node.js v12.3.1.
Type ".help" for more information.
> const Koa = require('koa')
undefined
> new Koa().callback()
[Function: handleRequest]
> 
```

Reply:
Yes, that’s how I explained it and that’s how I typed it? `callback()` is a method signature. It returns a function type `(req, res) => Promise<void>`.

Reply:
ahh I see, my bad, sorry about that